probe-rs/src/core/mod.rs

```diff
@@ -117,14 +119,75 @@ pub(crate) enum RegisterKind {
 /// Creating a new `RegisterValue` should be done using From or Into.
 /// Converting a value back to a primitive type can be done with either
 /// a match arm or TryInto
-#[derive(Debug, Clone, Copy, PartialEq)]
+#[derive(Debug, Clone, Copy)]
 pub enum RegisterValue {
     /// 32-bit unsigned integer
     U32(u32),
     /// 64-bit unsigned integer
     U64(u64),
 }
 
+impl RegisterValue {
+    /// A helper function to determine if the contained register value is equal to the maximum value that can be stored in that datatype.
+    pub fn is_max_value(&self) -> bool {
+        match self {
+            RegisterValue::U32(register_value) => *register_value == u32::MAX,
+            RegisterValue::U64(register_value) => *register_value == u64::MAX,
+        }
+    }
+
+    /// A helper function to determine if the contained register value is zero.
+    pub fn is_zero(&self) -> bool {
+        match self {
+            RegisterValue::U32(register_value) => register_value.is_zero(),
+            RegisterValue::U64(register_value) => register_value.is_zero(),
+        }
+    }
+}
+
+impl Default for RegisterValue {
```

Comment:
I'm curious - what are we using default for?

Reply:
To make it easy to 'unwrap_or_default()' in logs, when I'd rather use the Display formatting, vs. the Debug.